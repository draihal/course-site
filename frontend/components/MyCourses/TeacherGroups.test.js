import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestTeacherGroups from './TeacherGroups'

const groups = [
  {
    "id": 1,
    "url": "http://127.0.0.1:8000/api/v1/education/groups/pyweb-2019-09-16/",
    "name": "PyWeb-2019-09-16",
    "slug": "pyweb-2019-09-16"
  }
];


describe('TeacherGroups', () => {
  it('render correctly TeacherGroups component', () => {
    const tree = renderer.create(<TestTeacherGroups groups={groups} />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render TeacherGroups correctly with given props', () => {
    const { queryByText } = render(<TestTeacherGroups groups={groups} />);
    expect(queryByText(/PyWeb-2019-09-16/i)).toBeTruthy();
  });

});