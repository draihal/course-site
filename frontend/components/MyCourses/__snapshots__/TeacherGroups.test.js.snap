// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherGroups render correctly TeacherGroups component 1`] = `
<div
  className="container"
>
  <h2>
    Группы
  </h2>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    >
      <div
        className="card"
      >
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            PyWeb-2019-09-16
          </h5>
          <p
            className="card-text"
          >
            Some text.
          </p>
          <a
            className="btn btn-secondary btn-md float-right"
            href="/inner-sanctum?name=pyweb-2019-09-16"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Перейти
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
